common/src/main/scala/net/psforever/objects/zones/ZonePopulationActor.scala

```diff
@@ -148,39 +182,53 @@ object ZonePopulationActor {
 
   /**
     * If the given `player` passes a condition check, add it to the list.
+    * Also, ensure that "this player" is not currently counted among the living.
     * @param player a `Player` object
+    * @param playerMap the mapping of `Avatar` objects to `Player` objects
     * @param corpseList a list of `Player` objects
-    * @return true, if the `player` was added to the list;
-    *         false, otherwise
+    * @return a `Tuple` of two flags;
+    *         the first is whether the player was turned into a corpse or not;
+    *         the second is whether the player was found in the zone before being turned into a corpse
     */
-  def CorpseAdd(player : Player, corpseList : ListBuffer[Player]) : Boolean = {
+  def CorpseAdd(player : Player, playerMap : TrieMap[Avatar, Option[Player]], corpseList : ListBuffer[Player]) : (Boolean, Boolean) = {
```

Comment:
Is it worth converting corpseList to a TrieMap also, since it's passed around a fair bit, just to negate the chance that at some point it might be modified by multiple threads?

Reply:
Perhaps, but what would the key be?  It can't be any specific information related to the user's avatar because he is allowed multiple corpses to the same player avatar.

Reply:
The GUID of the corpse perhaps? As long as they're cleaned up properly there shouldn't be any conflicts.